src/test/munit/parameters/query-consistency/query-consistency-validation-test-suite.xml

```diff
@@ -130,5 +130,157 @@
+		</munit:validation>
+	</munit:test>
+
+	<munit:test name="validateNonNullableUnionArraysParams-QueryString">
+		<munit:enable-flow-sources>
+			<munit:enable-flow-source value="query-consistent-validation-main"/>
+			<munit:enable-flow-source value="get:\testQueryString:query-consistent-validation-config"/>
+		</munit:enable-flow-sources>
+		<munit:execution>
+			<http:request method="GET" config-ref="http-requester-simple"
+						  path="/api/testQueryString?nonNullableUnionOfArraysParams">
+				<http:response-validator>
+					<http:success-status-code-validator values="1..500"/>
+				</http:response-validator>
+			</http:request>
+		</munit:execution>
+		<munit:validation>
+			<munit-tools:assert-that expression="#[attributes.statusCode]" is="#[MunitTools::equalTo(400)]"/>
+		</munit:validation>
+	</munit:test>
+
+	<munit:test name="validateNonNullableUnionArraysParams-QueryParams">
+		<munit:enable-flow-sources>
+			<munit:enable-flow-source value="query-consistent-validation-main"/>
+			<munit:enable-flow-source value="get:\testQueryParams:query-consistent-validation-config"/>
+		</munit:enable-flow-sources>
+		<munit:execution>
+			<http:request method="GET" config-ref="http-requester-simple"
+						  path="/api/testQueryParams?nonNullableUnionOfArraysParams">
+				<http:response-validator>
+					<http:success-status-code-validator values="1..500"/>
+				</http:response-validator>
+			</http:request>
+		</munit:execution>
+		<munit:validation>
+			<munit-tools:assert-that expression="#[attributes.statusCode]" is="#[MunitTools::equalTo(400)]"/>
+		</munit:validation>
+	</munit:test>
+
+	<munit:test name="validateNonNullableUnionNullableArraysParams-QueryString">
+		<munit:enable-flow-sources>
+			<munit:enable-flow-source value="query-consistent-validation-main"/>
+			<munit:enable-flow-source value="get:\testQueryString:query-consistent-validation-config"/>
+		</munit:enable-flow-sources>
+		<munit:execution>
+			<http:request method="GET" config-ref="http-requester-simple"
+						  path="/api/testQueryString?nonNullableUnionOfNullableArraysParams">
+				<http:response-validator>
+					<http:success-status-code-validator values="1..500"/>
+				</http:response-validator>
+			</http:request>
+		</munit:execution>
+		<munit:validation>
+			<munit-tools:assert-that expression="#[attributes.statusCode]" is="#[MunitTools::equalTo(200)]"/>
+		</munit:validation>
+	</munit:test>
+
+	<munit:test name="validateNonNullableUnionNullableArraysParams-QueryParams">
+		<munit:enable-flow-sources>
+			<munit:enable-flow-source value="query-consistent-validation-main"/>
+			<munit:enable-flow-source value="get:\testQueryParams:query-consistent-validation-config"/>
+		</munit:enable-flow-sources>
+		<munit:execution>
+			<http:request method="GET" config-ref="http-requester-simple"
+						  path="/api/testQueryParams?nonNullableUnionOfNullableArraysParams=null">
```

Comment:
Although `=null` is the same as nothing, I would leave this one as the rest (without it) for consistency between test cases